c7n/query.py

```diff
@@ -91,6 +91,11 @@ def get(self, resource_manager, identities):
         """Get resources by identities
         """
         m = self.resolve(resource_manager.resource_type)
+        if isinstance(resource_manager, ChildResourceManager):
+            if getattr(m, 'supports_trailevents', False):
+                identities.append("trailevent")
```

Comment:
Looking for a better/more elegant way of doing this...

Reply:
its unclear why its being done at all tbh